packages/openapi-generator/src/parser/schema.ts

```diff
@@ -221,8 +229,11 @@ function parseXOfSchema(
   xOf: 'oneOf' | 'allOf' | 'anyOf',
   options: ParserOptions
 ): any {
+  const required = schema.required;
   return {
-    [xOf]: (schema[xOf] || []).map(entry => parseSchema(entry, refs, options))
+    [xOf]: (schema[xOf] || []).map(entry =>
+      parseSchema({ ...entry, required }, refs, options)
```

Comment:
[req] In this case the schema would overwrite anything that is required in the actual entry. We should at least merge this, but I think we shouldn't do this at all.

Reply:
Removed this for now. Required will be considered in a followup

Reply:
I merge this now. Have also added a test for it.